src/commands/heroku.rs

```diff
@@ -21,26 +21,29 @@ struct HerokuApp {
 #[command]
 #[num_args(1)]
 pub fn get_app(ctx: &mut Context, msg: &Message, mut args: Args) -> CommandResult {
-    let config = bot_config(ctx);
-
     let app_name = args
         .single::<String>()
         .expect("You must include an app name");
 
-    let response = heroku_client(&config.heroku_api_key)
-        .get()
-        .apps()
-        .app_name(&app_name)
-        .execute::<HerokuApp>();
+    let ctx_clone = ctx.clone();
+    let data = ctx_clone.data.read();
+
+    let heroku_client = data
+        .get::<HerokuClient>()
+        .expect("Expected Heroku client")
+        .clone();
```

Comment:
I don't think `.clone()` is doing anything here, since `HerokuClient` does not implement `Clone`, so this will just copy the reference.

Additionally, we may be able to avoid the code duplication by wrapping the `HerokuClient` in an `Arc` as we did for `Config` as well.

Reply:
Yeah, I'd love to see a `get_heroku_client(ctx)` function like `bot_config`.

Reply:
@pietroalbini Done!